### PR TITLE
Use datatypes and facts hash. Remove logic for Puppet 3.x

### PR DESCRIPTION
**Pull request by treydock**

### CODE REVIEW

**Comment by sheenaajay** [comment]
Thanks for submitting the PR @treydock .Could you please update the branch. Thank you.

**Comment by treydock** [comment]
@sheenaajay I've updated tests to pass

**Comment by sheenaajay** [comment]
@treydock Could you please rebase with the latest master. Thank you.

**Comment by treydock** [comment]
Rebased

**Comment by carabasdaniel** [comment]
Hello @treydock, thank you for your contribution, this looks good

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -152,15 +152,15 @@
       end
     end
 
-    ['string', false, ''].each do |param_value|
+    ['string', false, '', '1', '30'].each do |param_value|
```
Comment (glennsarti) [inline spec/classes/init_spec.rb]: Hrmm, this is a breaking change.

File: spec/classes/init_spec.rb [hunk 2]
```diff
@@ -53,7 +53,7 @@
       end
     end
 
-    invalid_url_values = ['\\\\ciflocation\\share', 'bob', '4', '']
```
Comment (glennsarti) [inline spec/classes/init_spec.rb]: This could potentially be breaking?

**Pull request merged**